### PR TITLE
#193 Support for open read silently

### PR DESCRIPTION
**Pull request by adrianchivu**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/pulls) open
- [X] I have verified that I am following the COMET-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [ ] I have provided test coverage for my change (where applicable)

### CODE REVIEW

File: CDP4Dal/ISession.cs
Comment (samatrhea) [inline CDP4Dal/ISession.cs:186]: my preference would be to add a parameter here that is not mandatory and true by default. Something like:

`Task Open(bool activeMessageBus = true);`

Then we do not need an extra method that in my opinion will cause confusion.

@alexatrhea : what do you think

File: CDP4Dal/Session.cs
Comment (samatrhea) [inline CDP4Dal/Session.cs:426]: same here, i would prefer:

`public async Task Read(Iteration iteration, DomainOfExpertise domain, bool activeMessageBus = true)`

File: CDP4Common/CDP4Common.csproj
Comment (samatrhea) [inline CDP4Common/CDP4Common.csproj:7]: i have the impression that nothing has changed in the CDP4Common project, so why bump the version? if nothing has changed, please revert

File: CDP4Dal/ISession.cs
Comment (samatrhea) [inline CDP4Dal/ISession.cs:198]: my preference would be to not have an extra method on this interface and implementation. Please use

`Task Open(bool activeMessageBus = true);`

Then we do not need an extra method that in my opinion will cause confusion.

File: CDP4Dal/ISession.cs
Comment (samatrhea) [inline CDP4Dal/ISession.cs:228]: same comment as above, not so happy with an extra method

File: CDP4Dal/Session.cs
Comment (samatrhea) [inline CDP4Dal/Session.cs:390]: please remove

File: CDP4JsonSerializer/CDP4JsonSerializer.csproj
Comment (samatrhea) [inline CDP4JsonSerializer/CDP4JsonSerializer.csproj:7]: i have the impression that nothing has changed in the CDP4JsonSerializerproject, so why bump the version? if nothing has changed, please revert

File: CDP4Rules/CDP4Rules.csproj
Comment (samatrhea) [inline CDP4Rules/CDP4Rules.csproj:7]: i have the impression that nothing has changed in CDP4Rules, please revert

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4Rules/CDP4Rules.csproj
Comment (adrianchivu) [reply CDP4Rules/CDP4Rules.csproj:7]: Got it. My thought was to increase the version of each of them(projects) in case of a new SDK release.

File: CDP4Common/CDP4Common.csproj
Comment (alexatrhea) [reply CDP4Common/CDP4Common.csproj:7]: ah there is a change im Common that we wanted to include also that got forgotten, so thats coming, and then the version bumps to all the other projects (because of dependency) are justified.

File: CDP4Dal/ISession.cs
Comment (alexatrhea) [reply CDP4Dal/ISession.cs:186]: yes agreed

**Review by alexatrhea — CHANGES_REQUESTED** [review]

File: CDP4Rules/CDP4Rules.csproj
Comment (alexatrhea) [reply CDP4Rules/CDP4Rules.csproj:7]: @samatrhea so now with the update of MessageBus in Common I think the version bumps to all libs are necessary.

File: CDP4Dal/CDPMessageBus.cs
Comment (samatrhea) [inline CDP4Dal/CDPMessageBus.cs:71]: i see this being introduced but the value does not seem to be set? please remove if not being used

File: CDP4Dal/CDPMessageBus.cs
Comment (samatrhea) [inline CDP4Dal/CDPMessageBus.cs:81]: i propose we rename this to "IsEnabled" to be inline with the Enable/Disable methods. if you do rename, make sure the backing field on line 76 is renamed as well/

Please rephrase docs using the COMET style for boolean:
```
Gets a value indicating whether the <see cref="CDPMessageBus"/> is active or not
```

Also curious why you choose to have a backing field, a read-only property and 2 methods to set the backing field. Why not use a property with a getter and a setter?

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4Dal/CDPMessageBus.cs
Comment (adrianchivu) [reply CDP4Dal/CDPMessageBus.cs:71]: It was there, used inside Listen method(I don't know why the viewer/differ marked this as added):
```
        public IObservable<T> Listen<T>(object target = null, string contract = null)
        {
            this.ListenerCallCount++;
            return this.GetOrAddObservable<T>(new EventTypeTarget(typeof(T), target));
        }
```

File: CDP4Dal/CDPMessageBus.cs
Comment (samatrhea) [inline CDP4Dal/CDPMessageBus.cs:76]: i think this needs to be public, no?

**Review by samatrhea — CHANGES_REQUESTED** [review]

**Review by samatrhea — APPROVED** [review]

**Comment by alexatrhea** [comment]
@adrianchivu lets wait for CI to run againm I added a small fix we found that broke coverage stats. After that we can merge and wait for CI again.

Once this is landed @samatrhea can we make a release so that the guys can proceed to make a new SAT package?

**Comment by samatrhea** [comment]
> 
> 
> @adrianchivu lets wait for CI to run againm I added a small fix we found that broke coverage stats. After that we can merge and wait for CI again.
> 
> Once this is landed @samatrhea can we make a release so that the guys can proceed to make a new SAT package?

yes please

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**
